core/services/keystore/keys/exportutils.go

```diff
@@ -0,0 +1,75 @@
+package keys
+
+import (
+	"encoding/json"
+
+	keystore "github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+type Encrypted interface {
+	GetCrypto() keystore.CryptoJSON
+}
+
+// EncryptedKeyExport represents a chain specific encrypted key
+type EncryptedKeyExport struct {
+	KeyType   string              `json:"keyType"`
+	PublicKey string              `json:"publicKey"`
+	Crypto    keystore.CryptoJSON `json:"crypto"`
+}
+
+func (x EncryptedKeyExport) GetCrypto() keystore.CryptoJSON {
+	return x.Crypto
+}
+
+// FromEncryptedJSON gets key from json and password
+func FromEncryptedJSON[E Encrypted, K any](
+	identifier string,
+	keyJSON []byte,
+	password string,
+	passwordFunc func(string) string,
```

Comment:
hmmm most prefixes follow the <keyTypeIdentifier>.ToLowerCase() + "key". the only exceptions:
* solana: uses `solkey` rather than `solanakey`
* starkent: uses `starkkey` rather than `starknetkey` (but this one is easily fixable since it's a new key)